### PR TITLE
feat: Support running end to end tests without postgres

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
Closes https://github.com/influxdata/influxdb_iox/issues/7285

I would like to make it easier to run run end to end  tests (see https://github.com/influxdata/influxdb_iox/issues/7285)

# Changes:
1. Update the catalog configuration to be entirely based on the `INFLUXDB_IOX_CATALOG_DSN` rather than DSN and `INFLUXDB_IOX_CATALOG_TYPE`.
2. Update the `influxdb_iox` end to end test harness to use the sqlite file based catalog if `TEST_INFLUXDB_IOX_CATALOG_DSN` is not set 

So the DSN can now be one of:
1. postgres (unchanged)
4. `memory` (memory)
5. `sqlite://<filename>`

# Example uses
Now you can run end to end tests like this (without a bunch of environment set) and it works great

```sql
cargo test --test end_to_end 
```

If you run it with only `TEST_INFLUXDB_IOX_CATALOG_DSN` you get:

```shell
TEST_INFLUXDB_IOX_CATALOG_DSN=postgresql://postgres@localhost:5432/postgres  cargo test --test end_to_end  -- flightsql_adhoc_query --nocapture

skipping end-to-end integration tests because TEST_INFLUXDB_IOX_CATALOG_DSN is set but TEST_INTEGRATION is not. To run, set TEST_INTEGRATION
skipping end-to-end integration tests because TEST_INFLUXDB_IOX_CATALOG_DSN is set but TEST_INTEGRATION is not. To run, set TEST_INTEGRATION
```

If you run it with just `TEST_INTEGRATION` you get:

```
TEST_INTEGRATION=1  cargo test --test end_to_end  -- flightsql_adhoc_query --nocapture

'end_to_end_cases::flightsql::flightsql_adhoc_query_errorend_to_end_cases::flightsql::flightsql_adhoc_query' panicked at '' panicked at 'TEST_INTEGRATION is set which requires running integration tests, but TEST_INFLUXDB_IOX_CATALOG_DSN is not set. Please set TEST_INFLUXDB_IOX_CATALOG_DSN to the test catalog database. For example, `TEST_INFLUXDB_IOX_CATALOG_DSN=postgres://postgres@localhost/iox_shared_test` would connect to a Postgres catalog.TEST_INTEGRATION is set which requires running integration tests, but TEST_INFLUXDB_IOX_CATALOG_DSN is not set. Please set TEST_INFLUXDB_IOX_CATALOG_DSN to the test catalog database. For example, `TEST_INFLUXDB_IOX_CATALOG_DSN=postgres://postgres@localhost/iox_shared_test` would connect to a Postgres catalog.', ', influxdb_iox/tests/end_to_end_cases/flightsql.rsinfluxdb_iox/tests/end_to_end_cases/flightsql.rs::7833::2424
```

### CODE REVIEW

**Comment by philjb** [comment]
I support this change. While the readme is very clear about how to use the env vars, I have to refer to the read each time. (ps don't forget to update the readme!)

**Review by crepererum — APPROVED** [review]
Updating docs required (as pointed out by others).

I really like the fusion of the catalog type and the DSN into a single parameter :+1: 

Ah, might be worth checking our K8s configs afterwards to remove `INFLUXDB_IOX_CATALOG_TYPE`.

File: clap_blocks/src/catalog_dsn.rs [hunk 1]
```diff
@@ -50,23 +50,16 @@ fn default_hotswap_poll_interval_timeout() -> &'static str {
 /// CLI config for catalog DSN.
 #[derive(Debug, Clone, Default, clap::Parser)]
 pub struct CatalogDsnConfig {
-    /// The type of catalog to use. "memory" is only useful for testing purposes.
-    #[clap(
-        value_enum,
-        long = "catalog",
-        env = "INFLUXDB_IOX_CATALOG_TYPE",
```
Comment (alamb) [inline clap_blocks/src/catalog_dsn.rs:57]: I double checked in k8s-idpe and it seems as if this environment variable is not set anywhere:

https://github.com/search?q=repo%3Ainfluxdata%2Fk8s-idpe%20INFLUXDB_IOX_CATALOG_TYPE&type=code

**Comment by alamb** [comment]
@philjb  and @crepererum  said:

> While the readme is very clear about how to use the env vars, I have to refer to the read each time. (ps don't forget to update the readme!)

> Updating docs required (as pointed out by others).

I have updated the content in `docs/testing.md` as part of this PR. Is that sufficient or did you have something else in mind?

File: docs/testing.md [hunk 2]
```diff
@@ -20,9 +20,16 @@ subsystem crate.
...
+The end to end tests are run using the `cargo test --test end_to_end`
+command. By default these tests run with a file based sqlite catalog.
+
+To ensure the tests run against an external catalog (such as posgres) you
+must set both the `TEST_INTEGRATION` and
+`TEST_INFLUXDB_IOX_CATALOG_DSN` environment variables.
+
+NOTE if you don't set `TEST_INTEGRATION` but do set
+`TEST_INFLUXDB_IOX_CATALOG_DSN` the tests will "pass" locally (really
+they will be skipped).
```
Comment (philjb) [inline docs/testing.md:32]: yes here and in crate iox_catalog/README.md - I think that's ok as is still?

Comment (alamb) [reply docs/testing.md:32]: Yes I just double checked that the instructions in `iox_catalog/README.md` are still valid 👍

**Comment by alamb** [comment]
This PR shows there are intermittent failures when using the local sqlite catalog -- for example 

https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/32774/workflows/2699244d-68ed-4186-9abd-8c2147c14d10/jobs/303635

https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/32769/workflows/a14b0f6c-fc24-44fd-aa10-3ab2a43d7fb7/jobs/303572

I don't want to unleash this on others until I figure out the flake

**Comment by alamb** [comment]
> I don't want to unleash this on others until I figure out the flake

Update here is that the flake seems to happen when concurrent processes try to update the same sqlite database file -- this is fine in theory -- we just need to wait until the other process unlocks the file

**Comment by crepererum** [comment]
> > I don't want to unleash this on others until I figure out the flake
> 
> Update here is that the flake seems to happen when concurrent processes try to update the same sqlite database file -- this is fine in theory -- we just need to wait until the other process unlocks the file

Doesn't sqllite has a mode where multiple processes can write-open the same database file? 🤔

**Comment by alamb** [comment]
> Doesn't sqllite has a mode where multiple processes can write-open the same database file? 🤔

I think so (wal mode, perhaps?) -- I started looking into it (aka figuring out how to configure sqlite) but haven't gotten far enough

File: iox_catalog/src/sqlite.rs [hunk 3]
```diff
@@ -198,9 +198,11 @@ impl TransactionFinalize for SqliteTxn {
 impl SqliteCatalog {
     /// Connect to the catalog store.
     pub async fn connect(options: SqliteConnectionOptions, metrics: Arc<Registry>) -> Result<Self> {
-        let opts = SqliteConnectOptions::from_str(&options.dsn)
+        let opts = SqliteConnectOptions::from_str(&options.file_path)
             .map_err(|e| Error::SqlxError { source: e })?
-            .create_if_missing(true);
+            .create_if_missing(true)
+            // use journaling mode to support concurrent writes
```
Comment (alamb) [inline iox_catalog/src/sqlite.rs:204]: > Doesn't sqllite has a mode where multiple processes can write-open the same database file?

I think by default sqlite support concurrent writes to the same file by different processes -- however, when I turn on  WAL mode the intermittent failure happens much more often.

WAL mode: https://www.sqlite.org/wal.html

I think what is happening is that the router creates the namespace so fast that the ingester doesn't see it -- I will leave more details in another comment

File: router/src/namespace_cache/read_through_cache.rs [hunk 4]
```diff
@@ -60,23 +61,12 @@ where
                 assert_eq!(cache_ns, *namespace);
                 let mut repos = self.catalog.repositories().await;
 
-                let schema = match get_schema_by_name(
+                let schema = get_schema_by_name_retry(
```
Comment (alamb) [inline router/src/namespace_cache/read_through_cache.rs:64]: I am not at all sure this is the right approach or place for this code. 

I would appreciate a review from @domodwyer  or @savage-engineer or @wiedld

If I revert the change in this file and run the following (without `TEST_INTEGRATION` set) the end to end tests fail (note file based tests are currently skipped on master)

```shell
# look! end to end tests without postgres!
cargo test --test end_to_end -- data_with_tags_no_predicate
```

It runs IOx like this (locally) with three processes reading / writing to the same shared file (sqlite database):
```
┌─────────────┐    ┌─────────────┐    ┌─────────────┐
│   Router    │    │  Ingester   │    │   Querier   │
└─────────────┘    └─────────────┘    └─────────────┘
       │                  │                  │       
       │                  │                  │       
       └──────────────────┼──────────────────┘       
                          │                          
                          │                          
                       .──▼──.                       
                      (       )   Shared sqlite      
                      │`─────'│   file database      
                      │       │  /tmp/db.sqlite      
                      │.─────.│                      
                      (       )                      
                       `─────'                       
```

The test fails (full output below) with a 500 INTERNAL error 

<details><summary>Full Test Output</summary>
<p>

```
---- end_to_end_cases::querier::influxrpc::metadata::tag_values::data_with_tags_no_predicate stdout ----
thread 'end_to_end_cases::querier::influxrpc::metadata::tag_values::data_with_tags_no_predicate' panicked at 'Invalid response code while writing line protocol:

Line Protocol:
h2o,state=CA,city=LA,county=LA temp=70.4 100
h2o,state=MA,city=Boston,county=Suffolk temp=72.4 250
o2,state=MA,city=Boston temp=50.4 200
o2,state=CA temp=79.0 300

Expected Status: 204 No Content
Actual Status: 500 Internal Server Error

Body:
b"{\"code\":\"internal error\",\"message\":\"failed to resolve namespace ID: namespace 5865378149202717_1660337837754476 not found\"}"', /Users/alamb/Software/influxdb_iox/test_helpers_end_to_end/src/steps.rs:315:21
stack backtrace:
   0: rust_begin_unwind
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
   1: core::panicking::panic_fmt
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
   2: test_helpers_end_to_end::steps::StepTest<S>::run::{{closure}}
             at /Users/alamb/Software/influxdb_iox/test_helpers_end_to_end/src/steps.rs:315:21
   3: end_to_end::end_to_end_cases::querier::influxrpc::InfluxRpcTest::run::{{closure}}
             at ./tests/end_to_end_cases/querier/influxrpc.rs:111:13
   4: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/future/future.rs:125:9
   5: end_to_end::end_to_end_cases::querier::influxrpc::metadata::tag_values::data_with_tags_no_predicate::{{closure}}
             at ./tests/end_to_end_cases/querier/influxrpc/metadata/tag_values.rs:70:5
   6: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/future/future.rs:125:9
   7: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/future/future.rs:125:9
   8: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}::{{closure}}
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:541:57
   9: tokio::runtime::coop::with_budget
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/coop.rs:107:5
  10: tokio::runtime::coop::budget
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/coop.rs:73:5
  11: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:541:25
  12: tokio::runtime::scheduler::current_thread::Context::enter
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:350:19
  13: tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:540:36
  14: tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:615:57
  15: tokio::macros::scoped_tls::ScopedKey<T>::set
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/macros/scoped_tls.rs:61:9
  16: tokio::runtime::scheduler::current_thread::CoreGuard::enter
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:615:27
  17: tokio::runtime::scheduler::current_thread::CoreGuard::block_on
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:530:19
  18: tokio::runtime::scheduler::current_thread::CurrentThread::block_on
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/scheduler/current_thread.rs:154:24
  19: tokio::runtime::runtime::Runtime::block_on
             at /Users/alamb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.28.0/src/runtime/runtime.rs:302:47
  20: end_to_end::end_to_end_cases::querier::influxrpc::metadata::tag_values::data_with_tags_no_predicate
             at ./tests/end_to_end_cases/querier/influxrpc/metadata/tag_values.rs:63:5
  21: end_to_end::end_to_end_cases::querier::influxrpc::metadata::tag_values::data_with_tags_no_predicate::{{closure}}
             at ./tests/end_to_end_cases/querier/influxrpc/metadata/tag_values.rs:62:40
  22: core::ops::function::FnOnce::call_once
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5
  23: core::ops::function::FnOnce::call_once
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
2023-04-28T19:42:54.455171Z  INFO clap_blocks::catalog_dsn: Catalog: Sqlite file_path="/var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpIjtOsu/MOIsYDlwLQebOPZApRnV.sqlite"
2023-04-28T19:42:54.457382Z  INFO clap_blocks::object_store: Object Store db_dir="/var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpGuW1Cg" object_store_type="Directory"
2023-04-28T19:42:54.459142Z  INFO router::dml_handlers::rpc_write::lazy_connector: connected to upstream ingester endpoint=http://127.0.0.1:8090/
2023-04-28T19:42:54.461585Z  INFO influxdb_iox::commands::run::router2: starting router2
2023-04-28T19:42:54.461888Z  INFO influxdb_iox::commands::run::main: InfluxDB IOx server starting git_hash=82922ea5eb07c22920559e3403798a6cace4178f version=0.1.0 uuid=6593354a-8835-430f-906f-d567ebeceb3f num_cpus=16 build_malloc_conf=
2023-04-28T19:42:54.462109Z  INFO influxdb_iox::commands::run::main: Binding gRPC services grpc_bind_address=SocketAddr(127.0.0.1:8093) server_type=RpcWriteRouter
2023-04-28T19:42:54.462229Z  INFO ioxd_common: bound gRPC listener local_addr=127.0.0.1:8093
2023-04-28T19:42:54.462257Z  INFO influxdb_iox::commands::run::main: Completed bind of gRPC, binding http http_bind_address=SocketAddr(127.0.0.1:8092) server_type=RpcWriteRouter
2023-04-28T19:42:54.462322Z  INFO ioxd_common: bound HTTP listener bind_addr=127.0.0.1:8092
2023-04-28T19:42:54.462378Z  INFO ioxd_common: gRPC server listening server_type=RpcWriteRouter
2023-04-28T19:42:54.462401Z  INFO ioxd_common: InfluxDB IOx RpcWriteRouter server ready    
2023-04-28T19:42:54.474240Z  INFO ioxd_common: HTTP server listening server_type=RpcWriteRouter
2023-04-28T19:42:54.649644Z  WARN router::namespace_cache::read_through_cache: failed to retrieve namespace schema error=namespace 5865378149202717_1660337837754476 not found namespace=5865378149202717_1660337837754476
2023-04-28T19:42:54.650455Z  WARN router::namespace_cache::read_through_cache: failed to retrieve namespace schema error=namespace 5865378149202717_1660337837754476 not found namespace=5865378149202717_1660337837754476
2023-04-28T19:42:54.650562Z ERROR ioxd_common::http: Error while handling request error=internal error: failed to resolve namespace ID: namespace 5865378149202717_1660337837754476 not found method=POST uri=/api/v2/write?org=5865378149202717&bucket=1660337837754476 content_length=Some("162")
2023-04-28T19:42:54.749889Z  INFO ioxd_common: Received SIGTERM
2023-04-28T19:42:54.749959Z  INFO ioxd_common: shutdown requested server_type=RpcWriteRouter
2023-04-28T19:42:54.750110Z  INFO ioxd_common: frontend shutdown completed server_type=RpcWriteRouter
2023-04-28T19:42:54.750134Z  INFO ioxd_common: backend shutdown completed server_type=RpcWriteRouter
2023-04-28T19:42:54.750170Z  INFO influxdb_iox::commands::run::main: done serving, draining futures grpc_bind_address=SocketAddr(127.0.0.1:8093) http_bind_address=Some(SocketAddr(127.0.0.1:8092)) server_type=RpcWriteRouter
2023-04-28T19:42:54.347756Z  INFO clap_blocks::catalog_dsn: Catalog: Sqlite file_path="/var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpIjtOsu/MOIsYDlwLQebOPZApRnV.sqlite"
2023-04-28T19:42:54.350668Z  INFO clap_blocks::object_store: Object Store db_dir="/var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpGuW1Cg" object_store_type="Directory"
2023-04-28T19:42:54.351612Z  INFO ingester2::persist::handle: initialised persist task n_workers=5 persist_queue_depth=250
2023-04-28T19:42:54.352076Z  INFO wal: Initalizing Write Ahead Log (WAL) wal_dir="/var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpz5o7Pl"
2023-04-28T19:42:54.394074Z  INFO ingester2::init::wal_replay: no wal replay files found
2023-04-28T19:42:54.394860Z  INFO influxdb_iox::commands::run::ingester2: starting ingester2
2023-04-28T19:42:54.395410Z  INFO influxdb_iox::commands::run::main: InfluxDB IOx server starting git_hash=82922ea5eb07c22920559e3403798a6cace4178f version=0.1.0 uuid=2061dc80-8c31-4dbc-9ccc-e4b7520e664a num_cpus=16 build_malloc_conf=
2023-04-28T19:42:54.399730Z  INFO influxdb_iox::commands::run::main: Binding gRPC services grpc_bind_address=SocketAddr(127.0.0.1:8090) server_type=Ingester2
2023-04-28T19:42:54.399969Z  INFO ioxd_common: bound gRPC listener local_addr=127.0.0.1:8090
2023-04-28T19:42:54.400018Z  INFO influxdb_iox::commands::run::main: Completed bind of gRPC, binding http http_bind_address=SocketAddr(127.0.0.1:8091) server_type=Ingester2
2023-04-28T19:42:54.400130Z  INFO ioxd_common: bound HTTP listener bind_addr=127.0.0.1:8091
2023-04-28T19:42:54.400222Z  INFO ioxd_common: gRPC server listening server_type=Ingester2
2023-04-28T19:42:54.400258Z  INFO ioxd_common: InfluxDB IOx Ingester2 server ready    
2023-04-28T19:42:54.413876Z  INFO ioxd_common: HTTP server listening server_type=Ingester2
2023-04-28T19:42:54.754316Z  INFO ioxd_common: Received SIGTERM
2023-04-28T19:42:54.754383Z  INFO ioxd_common: shutdown requested server_type=Ingester2
2023-04-28T19:42:54.754464Z  INFO ingester2::init::graceful_shutdown: gracefully stopping ingester
2023-04-28T19:42:54.754487Z  INFO ingester2::init::graceful_shutdown: persisting all data before shutdown
2023-04-28T19:42:54.774517Z  INFO ingester2::init::graceful_shutdown: persisted all data - stopping ingester
2023-04-28T19:42:54.774762Z  INFO ioxd_common: HTTP server shutdown server_type=Ingester2
2023-04-28T19:42:54.774813Z  INFO ioxd_common: frontend shutdown completed server_type=Ingester2
2023-04-28T19:42:54.774965Z  INFO ioxd_common: backend shutdown completed server_type=Ingester2
2023-04-28T19:42:54.775666Z  INFO influxdb_iox::commands::run::main: done serving, draining futures grpc_bind_address=SocketAddr(127.0.0.1:8090) http_bind_address=Some(SocketAddr(127.0.0.1:8091)) server_type=Ingester2
2023-04-28T19:42:54.776514Z  WARN executor: DedicatedExecutor dropped without calling shutdown()
2023-04-28T19:42:54.776604Z  WARN executor: DedicatedExecutor dropped without waiting for worker termination
2023-04-28T19:42:54.776946Z  WARN executor: DedicatedExecutor dropped without calling shutdown()
2023-04-28T19:42:54.777002Z  WARN executor: DedicatedExecutor dropped without waiting for worker termination
2023-04-28T19:42:54.567255Z  INFO clap_blocks::catalog_dsn: Catalog: Sqlite file_path="/var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpIjtOsu/MOIsYDlwLQebOPZApRnV.sqlite"
2023-04-28T19:42:54.569608Z  INFO clap_blocks::object_store: Object Store db_dir="/var/folders/s3/h5hgj43j0bv83shtmz_t_w400000gn/T/.tmpGuW1Cg" object_store_type="Directory"
2023-04-28T19:42:54.570033Z  INFO influxdb_iox::commands::run::querier: using specified number of threads per thread pool num_threads=4
2023-04-28T19:42:54.570065Z  INFO influxdb_iox::commands::run::querier: using ingester addresses ingester_addresses=[IngesterAddress { uri: http://127.0.0.1:8090/ }]
2023-04-28T19:42:54.573260Z  INFO influxdb_iox::commands::run::querier: starting querier
2023-04-28T19:42:54.573575Z  INFO influxdb_iox::commands::run::main: InfluxDB IOx server starting git_hash=82922ea5eb07c22920559e3403798a6cace4178f version=0.1.0 uuid=01ca5f8e-08d6-422d-bb28-d7759afe3ac6 num_cpus=16 build_malloc_conf=
2023-04-28T19:42:54.573870Z  INFO influxdb_iox::commands::run::main: Binding gRPC services grpc_bind_address=SocketAddr(127.0.0.1:8096) server_type=Querier
2023-04-28T19:42:54.574065Z  INFO ioxd_common: bound gRPC listener local_addr=127.0.0.1:8096
2023-04-28T19:42:54.574098Z  INFO influxdb_iox::commands::run::main: Completed bind of gRPC, binding http http_bind_address=SocketAddr(127.0.0.1:8095) server_type=Querier
2023-04-28T19:42:54.574180Z  INFO ioxd_common: bound HTTP listener bind_addr=127.0.0.1:8095
2023-04-28T19:42:54.574247Z  INFO ioxd_common: gRPC server listening server_type=Querier
2023-04-28T19:42:54.574273Z  INFO ioxd_common: InfluxDB IOx Querier server ready    
2023-04-28T19:42:54.586080Z  INFO ioxd_common: HTTP server listening server_type=Querier
2023-04-28T19:42:54.781319Z  INFO ioxd_common: Received SIGTERM
2023-04-28T19:42:54.781376Z  INFO ioxd_common: shutdown requested server_type=Querier
2023-04-28T19:42:54.781630Z  INFO ioxd_common: frontend shutdown completed server_type=Querier
2023-04-28T19:42:54.781690Z  INFO ioxd_common: backend shutdown completed server_type=Querier
2023-04-28T19:42:54.781783Z  INFO influxdb_iox::commands::run::main: done serving, draining futures grpc_bind_address=SocketAddr(127.0.0.1:8096) http_bind_address=Some(SocketAddr(127.0.0.1:8095)) server_type=Querier


failures:
    end_to_end_cases::querier::influxrpc::metadata::tag_values::data_with_tags_no_predicate

test result: FAILED(B. 0 passed; 1 failed; 0 ignored; 0 measured; 271 filtered out; finished in 2.55s
```


</p>
</details> 


I believe the error comes from the the ingester process not being able to find the namespace entry in the catalog, despite it having been put there by the router. You can see this in the following message:


```
2023-04-28T19:42:54.649644Z  WARN router::namespace_cache::read_through_cache: failed to retrieve namespace schema error=namespace 5865378149202717_1660337837754476 not found namespace=5865378149202717_1660337837754476
```

I suspect the router/ingester are so fast 🚀 , the sqlite instance on the ingester doesn't see the update until after the ingester has gotten the write request

**Comment by alamb** [comment]
🤔  here is another failure related to concurrent writes to the same sqlite database: https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/32908/workflows/e88630de-4f32-4758-827c-809175e2e238/jobs/305141 (we just need to automatically retry I think)

**Comment by alamb** [comment]
After some thought last night, my current plan for this PR is to factor out the changes for being able to run end to end tests with sqlite but not actually make that the default

Then we can iterate on how to get the tests to pass without flake with sqlite backed tests separately

**Comment by alamb** [comment]
Update here: 
1. https://github.com/influxdata/influxdb_iox/pull/7708 makes it possible to run the end to end tests with sqlite
2. https://github.com/influxdata/influxdb_iox/issues/7709 tracks work to get the tests to run using sqlite by default and in CI

Closing this PR for now

**Pull request closed without merging**